gnomad_qc/v4/assessment/calculate_per_sample_stats.py

```diff
@@ -427,18 +436,40 @@ def get_summary_stats_filter_groups_ht(
             filter_lcr=True,
             filter_expr=ht.filters,
             freq_expr=ht.freq[0].AF,
+            grpmax_expr=grpmax_expr,
             max_af=rare_variants_afs,
+            max_grpmax=rare_variants_grpmax,
         ),
         **csq_filter_expr,
     }
 
     # Create the metadata for all requested filter groups.
     ss_filters = deepcopy(SUM_STAT_FILTERS)
-    ss_filters["max_af"] = rare_variants_afs
+    if rare_variants_afs:
+        ss_filters["max_af"] = rare_variants_afs
+    if rare_variants_grpmax:
+        ss_filters["max_grpmax"] = rare_variants_grpmax
+
+    # If rare_variant_mode is True, then only include the rare variant filter groups
+    # and variant_qc as the common filter groups.
+    common_filters = deepcopy(COMMON_FILTERS)
+    common_filter_combos = deepcopy(COMMON_FILTER_COMBOS)
+    if rare_variant_mode:
+        common_filter_combos = []
+        rare_variant_max = {}
+        if rare_variants_afs:
+            rare_variant_max["max_af"] = max(rare_variants_afs)
+            common_filters["max_af"] = rare_variants_afs
+            common_filter_combos.append(["variant_qc", "max_af"])
+        if rare_variants_grpmax:
+            rare_variant_max["max_grpmax"] = max(rare_variants_grpmax)
+            common_filters["max_grpmax"] = rare_variants_grpmax
+            common_filter_combos.append(["variant_qc", "max_grpmax"])
```

Comment:
I can see where we might only care about one or the other. I'm including `max_grpmax` because of the plots we are thinking about making and `max_af` in case we want more granularity on the rare variants. Each adds a lot of stratifications, so just thought it would be nice to have the option to only do one of them if they are not both needed.

Reply:
I added an error if rare_variant_mode is True and both are None